x-pack/plugins/fleet/server/services/epm/elasticsearch/template/template.ts

```diff
@@ -193,7 +196,6 @@ export function generateMappings(fields: Field[]): IndexTemplateMappings {
             break;
           default: {
             const meta = {};
-            if ('metric_type' in field) Reflect.set(meta, 'metric_type', field.metric_type);
```

Comment:
> question: can `metric_type` just be removed from `meta` or does it need to be kept/migrated for BWC?

I don't know of anything using the meta field, but maybe we can keep it for some time just in case? On this comment https://github.com/elastic/kibana/issues/82273#issuecomment-862304253 @ruflin also seemed to prefer keeping both approaches during some time.

> it looks like the change of adding time_series_metric causes errors when installing system or apm package:

Regarding the error, could it be that `aggregations` is mandatory when using the "object" format? In the examples in https://github.com/elastic/elasticsearch/issues/74014, this format is only used when `aggregations` is also present.

Maybe we have to use this format when there are no aggregations as is the case now:
```
              "time_series_metric": "gauge"
```

@csoulios is it ok to set only the metric type without aggregations?

Reply:
Thanks, it works if I change to string like `"time_series_metric": "gauge"`. If later we need to support aggregations, is it going to be possible to change from string to object?

Reply:
Currently, the `time_series_metric` field only supports a string value that can be any of the `gauge`, `counter`, `histogram`, `summary`.

I would like to ask which of the above values do you plan to use.

See https://github.com/elastic/elasticsearch/pull/76766

Reply:
I'm not sure what are all the possible values of metric_type/time_series_metric, maybe @jsoriano knows?
It looks like the string format works, so we can go ahead with that, let me know if any concerns.

Reply:
By now we only allow `counter` and `gauge` in packages.